File: backend/app/routes/templates.py
"""Routes for architecture templates."""

from fastapi import APIRouter, HTTPException
from app.data.templates import TEMPLATES

router = APIRouter(prefix="/api/templates", tags=["templates"])


@router.get("/")
async def get_templates():
    """
    Get all predefined architecture templates.
    
    Returns all available templates that users can use as starting points.
    """
    try:
        return TEMPLATES
    except Exception as e:
        raise HTTPException(status_code=500, detail="Failed to retrieve templates")


@router.get("/{template_name}")
async def get_template(template_name: str):
    """
    Get a specific template by name.
    
    Args:
        template_name: The name of the template to retrieve
        
    Returns:
        The requested template or 404 if not found
    """
    try:
        template = TEMPLATES.get(template_name)
        if not template:
            raise HTTPException(status_code=404, detail=f"Template '{template_name}' not found")
        return template
    except HTTPException:
        raise
    except Exception as e:
        raise HTTPException(status_code=500, detail="Failed to retrieve template")

File: backend/src/routes/templates.js
// Routes for architecture templates
const express = require('express');
const router = express.Router();
const TEMPLATES = require('../data/templates');

/**
 * GET /api/templates
 * Retrieve all predefined architecture templates
 */
router.get('/', (req, res) => {
  try {
    res.json(TEMPLATES);
  } catch (error) {
    res.status(500).json({ error: 'Failed to retrieve templates' });
  }
});

/**
 * GET /api/templates/:name
 * Retrieve a specific template by name
 */
router.get('/:name', (req, res) => {
  try {
    const templateName = req.params.name;
    const template = TEMPLATES[templateName];
    
    if (!template) {
      return res.status(404).json({ error: `Template '${templateName}' not found` });
    }
    
    res.json(template);
  } catch (error) {
    res.status(500).json({ error: 'Failed to retrieve template' });
  }
});

module.exports = router;
